backend/controllers/auth_controller.go

```diff
@@ -0,0 +1,92 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/ZPI-2024-25/KubernetesAccessManager/auth"
+	"github.com/ZPI-2024-25/KubernetesAccessManager/models"
+	"github.com/golang-jwt/jwt/v4"
+)
+
+func CheckLoginStatus(w http.ResponseWriter, r *http.Request)  {
+	token, err := auth.GetJWTTokenFromHeader(r)
+	isValid, claims := auth.IsTokenValid(token)
+
+	if err != nil || !isValid {
+		writeJSONResponse(w, http.StatusUnauthorized, models.ModelError{
+			Message: "Unauthorized",
+			Code: http.StatusUnauthorized,
+		})
+		return
+	}
+	rolemap, err := auth.GetRoleMapInstance()
+	if err != nil {
+		writeJSONResponse(w, http.StatusInternalServerError, &models.ModelError{
+			Message: "Internal Server Error",
+			Code: http.StatusInternalServerError,
+		})
+	}
+	status, errM := getLoginStatus(claims, rolemap)
+	if errM != nil {
+		writeJSONResponse(w, int(errM.Code), err)
+		return
+	}
+	writeJSONResponse(w, http.StatusOK, status)
+}
+
+func getLoginStatus(claims *jwt.MapClaims, rolemap *auth.RoleMapRepository) (*models.UserStatus, *models.ModelError) {
+	roles, err := auth.ExtractRoles(claims)
+	if err != nil {
+		return nil, &models.ModelError{
+			Message: "Roles not found in bearer token",
+			Code: http.StatusInternalServerError,
```

Comment:
This is for @samuelus, but shouldn't this be some other error than 500?

Reply:
I was thinking of bad request code 400, but api spec doesn't mention it so I decided to do it according to spec

Reply:
If you need some changes in API create ticket on Jira and assign it to me. Error 400 would actually fit more here, but error 500 will be also acceptable

Reply:
I don’t think 500 makes sense here. A 500 error indicates a server-side issue, but if the token is invalid, it’s a client-side error. So, it shouldn’t be 500.

Reply:
Changed it to bad request, ticket for you @samuelus https://samuelus.atlassian.net/browse/ZPI-105